docs/development/index.rst

```diff
@@ -447,35 +447,64 @@ In this case, use
 at the top of the file and then just use ``assert_quantity_allclose``
 for the tests.
 
+.. _development_random:
+
 Random numbers
 --------------
 
-When generating random numbers it is useful to define a `random
-state` (a.k.a. `seed`) in order to produce pseudo-random numbers.
-This has the advantage of having random number that are reproducible
-if using the same `random state`, so the results can be reproduced.
-
-In Gammapy, there is a utility function to deal with random states:
-`~gammapy.utils.random.check_random_state`. The best way is to
-import it on top of the file that should use it with
+All functions that need to call a random number generator should
+take a ``random_state`` input parameter and call the
+`~gammapy.utils.random.get_random_state` utility function like this
+(you can copy & paste the three docstring lines and the first code line
+to the function you're writing):
 
 .. code-block:: python
 
-    from gammapy.utils.random import check_random_state
+    from gammapy.utils.random import get_random_state
+
+    def make_random_stuff(X, random_state='random-seed'):
+        """...
 
-and define a random number generator as
+        Parameters
+        ----------
+        random_state : {int, 'random-seed', 'global-rng', `~numpy.random.RandomState`}
+            Defines random number generator initialisation.
+            Passed to `~gammapy.utils.random.get_random_state`.
+        """
+        random_state = get_random_state(random_state)
+        data = random_state.uniform(low=0, high=3, size=10)
+        return data
+
+This allows callers flexible control over which random number generator
+(i.e. which `numpy.random.RandomState` instance) is used and how it's initialised.
+The default ``random_state='random-seed'`` means "create a new RNG, seed it in a random way",
```

Comment:
@mapazarr, @adonath – Do you agree this is a good default?
Or do you think `seed=0` or using the global numpy random number generator by default would be better (and if so, why)?